pkg/gcs/sysfunc.go

```diff
@@ -256,11 +256,11 @@ func (e *Eval) setTargetPos(c *ast.CallExpr, env *Env) (Obj, error) {
 	}
 
 	//check if index is in range
-	if idx < 1 || idx >= e.Core.Combat.TargetsCount() {
-		return nil, fmt.Errorf("index for set_default_target is invalid, should be between %v and %v, got %v", 1, e.Core.Combat.TargetsCount()-1, idx)
+	if idx < 1 || idx >= e.Core.Combat.EnemyCount() {
+		return nil, fmt.Errorf("index for set_default_target is invalid, should be between %v and %v, got %v", 1, e.Core.Combat.EnemyCount()-1, idx)
```

Comment:
I don't think this should have a `-1`.

Reply:
Code here is actually wrong, should be

```go
	if idx < 1 || idx > e.Core.Combat.EnemyCount() {
		return nil, fmt.Errorf("index for set_default_target is invalid, should be between %v and %v, got %v", 1, e.Core.Combat.EnemyCount(), idx)
	}

	e.Core.Combat.SetEnemyPos(idx-1, x, y)
```

Used to be target[0] was Player so SetTarget (enemy really) should be 1+; However, now that it's independent enemy array, first enemy is actually 0 now, but for syntax consistency we should still allow 1 to len(...)